circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/RememberRetained.kt

```diff
@@ -84,7 +79,19 @@ import androidx.compose.runtime.saveable.rememberSaveable
  * @param init A factory function to create the initial value of this state
  */
 @Composable
-fun <T : Any> rememberRetained(vararg inputs: Any?, key: String? = null, init: () -> T): T {
+public fun <T : Any> rememberRetained(vararg inputs: Any?, key: String? = null, init: () -> T): T {
+  val registry = LocalRetainedStateRegistryOwner.current
+  // Short-circuit no-ops
+  if (registry === NoOpRetainedStateRegistry) {
+    return when (inputs.size) {
+      0 -> remember(init)
+      1 -> remember(inputs[0], init)
+      2 -> remember(inputs[0], inputs[1], init)
+      3 -> remember(inputs[0], inputs[1], inputs[2], init)
+      else -> remember(keys = inputs, init)
+    }
```

Comment:
Is there some benefit to manually handling 0-3 here as opposed to just doing the following in all cases?
```
return remember(keys = inputs, init)
```

Reply:
That always makes a copy of the array, whereas passing to overloads avoids it in the common case

Reply:
Ahh ok. I wondered if there was a performance win I was missing